# From UNINIT to TRANSITION_STATE_PRE_CAL
uint8 TRANSITION_CALIBRATE = 4


# From TRANSITION_STATE_PRE_CAL to IDLE
uint8 TRANSITION_CALIBRATE_COMPLETE = 2


# From IDLE to TRANSITION_STATE_ACTIVATING_POS_CONTROL
uint8 TRANSITION_ACTIVATE_POS_CONTROL = 4


# From IDLE to TRANSITION_STATE_ACTIVATING_VEL_CONTROL
uint8 TRANSITION_ACTIVATE_VEL_CONTROL = 6


# From IDLE/POS_CONTROL/VEL_CONTROL to TRANSITION_STATE_SHUTTING_DOWN
uint8 TRANSITION_SHUTDOWN = 8


# From TRANSITION_STATE_ACTIVATING_POS_CONTROL to POS_CONTROL
uint8 TRANSITION_ACTIVATE_POS_CONTROL_COMPLETE = 10


# From TRANSITION_STATE_ACTIVATING_VEL_CONTROL to VEL_CONTROL
uint8 TRANSITION_ACTIVATE_VEL_CONTROL_COMPLETE = 12


# From TRANSITION_STATE_SHUTTING_DOWN to FINALIZED
uint8 TRANSITION_SHUTDOWN_COMPLETE = 14


# From POS_CONTROL to TRANSITION_STATE_DEACTIVATING_POS_CONTROL
uint8 TRANSITION_DEACTIVATE_POS_CONTROL = 16


# From VEL_CONTROL to TRANSITION_STATE_DEACTIVATING_VEL_CONTROL
uint8 TRANSITION_DEACTIVATE_VEL_CONTROL = 18


# From TRANSITION_STATE_DEACTIVATING_POS_CONTROL to IDLE
uint8 TRANSITION_DEACTIVATE_POS_CONTROL_COMPLETE = 20


# From TRANSITION_STATE_DEACTIVATING_VEL_CONTROL to IDLE
uint8 TRANSITION_DEACTIVATE_VEL_CONTROL_COMPLETE = 22


# From TRANSITION_STATE_PRE_CAL to TRANSITION_STATE_ERR_PROCESSING
uint8 TRANSITION_CALIBRATE_ERROR = 99


# From TRANSITION_STATE_ACTIVATING_POS_CONTROL to TRANSITION_STATE_ERR_PROCESSING
uint8 TRANSITION_ACTIVATE_POS_CONTROL_ERROR = 99


# From TRANSITION_STATE_DEACTIVATING_POS_CONTROL to TRANSITION_STATE_ERR_PROCESSING
uint8 TRANSITION_DEACTIVATE_POS_CONTROL_ERROR = 99


# From TRANSITION_STATE_ACTIVATING_VEL_CONTROL to TRANSITION_STATE_ERR_PROCESSING
uint8 TRANSITION_ACTIVATE_VEL_CONTROL_ERROR = 99


# From TRANSITION_STATE_DEACTIVATING_VEL_CONTROL to TRANSITION_STATE_ERR_PROCESSING
uint8 TRANSITION_DEACTIVATE_VEL_CONTROL_ERROR = 99


# From TRANSITION_STATE_SHUTTING_DOWN to TRANSITION_STATE_ERR_PROCESSING
uint8 TRANSITION_SHUTDOWN_ERROR = 99


# From TRANSITION_STATE_ERR_PROCESSING to UNINIT
uint8 TRANSITION_ERR_PROCESSING_COMPLETE = 99


# From TRANSITION_STATE_ERR_PROCESSING to FINALIZED
uint8 TRANSITION_ERR_PROCESSING_ERROR = 99




# Transition result codes
uint8 TRANSITION_SUCCESS = 97
uint8 TRANSITION_FAILURE = 98
uint8 TRANSITION_ERROR = 99

# Transition request fields
uint8 id
string label